### PR TITLE
Creates UI for signIn page integrating with OAuth

### LINKED ISSUE #11: Adapt login to new app
* Implement same login architecture as cboard for this new app
* Create the Login/SignUp page
* If user is not logged in, they should be redirected to the Login/SignUp page
* Consider using [NextAuth/AuthJS](https://next-auth.js.org/)

image
image
image
image
image
image

### PR DESCRIPTION
**Pull request by maxicapodacqua**
Closes #11 

Things to fix in future iterations:
1. Layout spaces (some padding might not match design)
2. Mobile is not including the image at the bottom
3. Privacy and Terms links are not going to any page and their location is incorrect, should be at the bottom of the page

Features:
* When user is not logged in, the app will redirect them to the signin/ page
* After login they will be redirected to the dashboard (or any callbackURL they had)
* Login Dialog allows for email/password login
* OAuth buttons allows for login with Google/Facebook/Apple login
* Error messages are shown as MUI Alert components
* Refactored `BrandIcon` into a SVG component so we can control it MUI SvgIcon better

### CODE REVIEW

File: src/app/[locale]/signin/Container.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                sx={{ pb: '32px' }}
+              >
+                <Typography variant="h2">{t('sloganTitle')}</Typography>
+                <Chip label={t('ai')} variant="filled" sx={styles.chipAI} />
+              </Stack>
+              <Typography variant="h5">{t('sloganSubtitle')}</Typography>
+            </Box>
+          </Box>
+          <Box>
+            <Image
```
Comment (RodriSanchez1) [inline src/app/[locale]/signin/Container.tsx:74]: We need to add a felixble image container

**Comment by tomivm** [comment]
The web console is showing these errors
`Warning: Failed prop type: Invalid prop `children` supplied to `ForwardRef(Box)`, expected a ReactNode.`
`Warning: Extra attributes from the server: data-new-gr-c-s-check-loaded,data-gr-ext-installed`

**Comment by maxicapodacqua** [comment]
> Warning: Extra attributes from the server: data-new-gr-c-s-check-loaded,data-gr-ext-installed

That warning it's[ because of the Gramarly extension](https://stackoverflow.com/questions/75337953/what-causes-nextjs-warning-extra-attributes-from-the-server-data-new-gr-c-s-c) (I have the same warning and extension installed) .

I'll work on the ForwardRef warning now

File: src/app/[locale]/signin/Container.tsx
Comment (tomivm) [inline src/app/[locale]/signin/Container.tsx:1]: Please Avoid using the name 'Container' for the files. you could use the name of the component. like 'Signin'

**Review by tomivm — CHANGES_REQUESTED** [review]

**Pull request merged**